app/models/appointment.rb

```diff
@@ -17,8 +16,16 @@ class Appointment < ApplicationRecord
     end
   end
 
+  validate :client_is_unchanged, on: [:update]
```

Comment:
Nit: Ruby convention would be to drop the `is` and append a `?`: `:client_unchanged?`